python/pyspark/sql/types.py

```diff
@@ -483,7 +483,8 @@ def __init__(self, fields=None):
             self.names = [f.name for f in fields]
             assert all(isinstance(f, StructField) for f in fields),\
                 "fields should be a list of StructField"
-        self._needSerializeAnyField = any(f.needConversion() for f in self)
+        self._needConversion = [f.needConversion() for f in self]
```

Comment:
I'd rename this to another, for example, `_needConversions` (or others if there are) and leave a comment here why we do this.